packages/frontend/src/hooks/usePositions.ts

```diff
@@ -511,45 +513,55 @@ export const usePnL = () => {
     setShortGain(_gain)
   }, [buyQuote.toString(), ethPrice.toString(), wethAmount.toString(), squeethAmount.toString()])
 
-  const currentShortDeposits = useMemo(() => {
-    if (positionType === PositionType.LONG) return []
-    let totalShortSqth = new BigNumber(0)
-    const result = []
-    for (let index = 0; index < transactions.length; index++) {
-      if (totalShortSqth.gte(squeethAmount)) break
-      if (
-        totalShortSqth.isLessThan(squeethAmount) &&
-        transactions[index].transactionType === TransactionType.MINT_SHORT
-      ) {
-        totalShortSqth = totalShortSqth.plus(transactions[index].squeethAmount)
-        result.push(transactions[index])
-      } else if (
-        totalShortSqth.isLessThan(squeethAmount) &&
-        transactions[index].transactionType === TransactionType.BURN_SHORT
-      ) {
-        totalShortSqth = totalShortSqth.minus(transactions[index].squeethAmount)
-      }
-    }
-    return result
-  }, [positionType, squeethAmount.toString(), transactions.length])
+  const currentShortDeposits = useAsyncMemo(
+    async () => await getCurrentShortDeposits({ positionType, squeethAmount, transactions, getBuyQuote }),
```

Comment:
Thinking we shld use finalSqueethAmount here instead of squeethAmount here, also need to take LPed amount into account, but there will be a loading issue, LPed amount is calculated in the last part, it will take a while to load

Reply:
As long as we show a loading state, I think its more important to have the accurate value

Reply:
The finalSqueeth should be the accurate value?